### ISSUE TITLE
Allow installing .NET patch versions side by side

### ISSUE THREAD

**Issue by cmeeren** [dotnet/sdk#30113]
I currently have 7.0.102 installed. For debugging purposes, I would like to install 7.0.101 and force that exact version using `global.json`. However, the installer refuses to install  7.0.101 because "another version of the product is already installed". To be able to install, I have to modify my VS installation to remove the .NET SDK altogether (which also removes other stuff I would like to not touch), remove versions installed through Chocolatey, etc.

For examples of problems where this kind of installation/pinning/debugging make sense, see e.g. https://github.com/dotnet/fsharp/issues/13678 and https://github.com/dotnet/fsharp/issues/14618.

**Comment by baronfel** [comment]
That's right - it's an explicit design goal of the microsoft-authored installers and packages that patch versions be automatically updated. This is partially to help with the automatic update story with systems like Microsoft Update/debian unattended updates - for security reasons path updates should remove the (presumed vulnerable) previous versions.

I would recommend the `dotnet-install` scripts as an interim solution.
